### PR TITLE
Integrate trivy setup automation

### PR DESCRIPTION
**Pull request by stephen-fox**
This change set updates "post-command" to run the trivy executable directly instead of using a Docker container. The path to the trivy executable is provided by the "pre-checkout" hook via an environment variable.

Due to limitations with the "bats" testing framework, I set the PATH environment variable to contain the trivy executable's parent directory. This is, frankly, half-ass... but I am unsure of a better alternative.

This commit also updates the post-command's tests to handle these changes.

### CODE REVIEW

File: hooks/post-command [hunk 1]
```diff
@@ -1,13 +1,5 @@
 #!/bin/bash
 
-# NOTE(jaosorior): This is duplicated in the tests.
-readonly TRIVY_DEFAULT_VERSION="0.31.3"
-export TRIVY_VERSION="${BUILDKITE_PLUGIN_TRIVY_VERSION:-$TRIVY_DEFAULT_VERSION}"
-export image="aquasec/trivy:${TRIVY_VERSION}"
-
-args=()
-fsargs=()
-
```
Comment (JAORMX) [inline hooks/post-command:10]: woo-hoo!

File: hooks/post-command [hunk 2]
```diff
@@ -59,6 +51,21 @@ docker_metadata_file_exists() {
...
+[[ -z "${TRIVY_EXE_PATH}" ]] \
+  && fail_with_message "TRIVY_EXE_PATH env. is unset or empty"
+
+[[ ! -f "${TRIVY_EXE_PATH}" ]] \
+  && fail_with_message "trivy exe file does not exist ('${TRIVY_EXE_PATH}')"
+
+# I would prefer to just execute the path pointed-to by this environment
+# variable - but that is incompatible with the testing framework. As a
+# result, we need to modify PATH to include the directory containing
+# the trivy executable.
```
Comment (JAORMX) [inline hooks/post-command:63]: wait, so we can't just execute `"${TRIVY_EXE_PATH}"` instead? so we wouldn't need to rely on `PATH`

File: tests/post-command.bats [hunk 3]
```diff
@@ -2,17 +2,14 @@
...
-export image="aquasec/trivy:${TRIVY_VERSION}"
+export TRIVY_EXE_PATH="$(mktemp)"
 
 # Uncomment the following line to debug stub failures
 # export BUILDKITE_AGENT_STUB_DEBUG=/dev/tty
 
 @test "fs scan of a test app" {
   # TODO(jaosorior): Change the exit code if we change the default
-  stub docker "run -v \"$PWD\":/workdir --rm $image fs --exit-code 0 --security-checks vuln,config /workdir : echo fs scan success"
+  stub trivy "fs --exit-code 0 --security-checks vuln,config . : echo fs scan success"
```
Comment (JAORMX) [inline tests/post-command.bats:12]: the test look a lot cleaner now!

File: hooks/post-command (hunk 2 above)
Comment (stephen-fox) [reply hooks/post-command:63]: We cannot use the terrible `stub` stuff if we use the file path :(

**Pull request merged**